src/meta/processors/CreateSpaceProcessor.cpp

```diff
@@ -22,20 +23,25 @@ void CreateSpaceProcessor::process(const cpp2::CreateSpaceReq& req) {
         return;
     }
     CHECK_EQ(Status::SpaceNotFound(), spaceRet.status());
-    auto ret = allHosts();
-    if (!ret.ok()) {
+    auto hosts = ActiveHostsManHolder::hostsMan()->getActiveHosts();
+    if (hosts.empty()) {
         LOG(ERROR) << "Create Space Failed : No Hosts!";
         resp_.set_code(cpp2::ErrorCode::E_NO_HOSTS);
         onFinished();
         return;
     }
     auto spaceId = autoIncrementId();
-    auto hosts = ret.value();
     auto spaceName = properties.get_space_name();
     auto partitionNum = properties.get_partition_num();
     auto replicaFactor = properties.get_replica_factor();
     VLOG(3) << "Create space " << spaceName << ", id " << spaceId;
-
+    if ((int32_t)hosts.size() < replicaFactor) {
+        LOG(ERROR) << "Not enough hosts existed for replica "
+                   << replicaFactor << ", hosts num " << hosts.size();
+        resp_.set_code(cpp2::ErrorCode::E_UNSUPPORTED);
```

Comment:
`Unsupport` is the meaning that this operation does not implement.

`Illegal_Argument` will be better ?

Reply:
Thrift will return its own failure if related method does not implement.  So i think UNSUPPORTED is ok in this case.